newsroom/push/publishing.py

```diff
@@ -298,3 +300,27 @@ async def publish_planning_into_event(self, planning: dict[str, Any]) -> Optiona
             # Replace the original document
             await service.update(agenda["_id"], agenda)
             return agenda["_id"]
+
+    async def publish_planning_related_events(self, planning: dict[str, Any], related_event: dict[str, Any]):
+        if not related_event.get("link_type") == "secondary":
+            return None
+
+        service = AgendaItemService()
+        event_id = related_event.get("_id")
+
+        event = await service.find_by_id(event_id)
+        agenda = event.to_dict()
+
+        if agenda.get("item_type") == "event" and planning and planning.get("guid") not in agenda["planning_ids"]:
+            agenda["planning_ids"].append(planning["guid"])
+
+        if event_id not in agenda["event_ids"]:
```

Comment:
This actually doesn't belong here on the Event, as the Event already knows it's own ID. Instead you can store the list of `event_ids` in a function on the Agenda [convert_planning_to_agenda_dict](https://github.com/superdesk/newsroom-core/blob/async/newsroom/agenda/agenda_service.py#L196).
You would add there:

```python
plan["event_ids"] = [link["_id"] for link in (planning_item.get("related_events") or [])]
```

This way the Planning item stores the list of Event IDs that it is linked to

Reply:
so in this case we have to add `event_ids` field in the [planning schema](https://github.com/superdesk/newsroom-core/blob/async/newsroom/types/agenda.py#L146) right ?

Reply:
but if we populate `events_ids` here [convert_planning_to_agenda_dict](https://github.com/superdesk/newsroom-core/blob/async/newsroom/agenda/agenda_service.py#L196). we got pydantic validation error